runtime/compiler/optimizer/TreeLowering.cpp

```diff
@@ -38,13 +38,401 @@ TR::TreeLowering::optDetailString() const throw()
+      //   |   astorei/awrtbari             |
+      //   |     aladd                      |
+      //   |       <array-reference>        |
+      //   |       index-offset-calculation |
+      //   |     <value-reference>          |
+      //   +--------------------------------+
+      //
+      // into
+      //   +--------------------------------+
+      //   | treetop                        |
+      //   |   <array-reference>            |
+      //   | treetop                        |
+      //   |   <value-reference>            |
+      //   | ificmpeq  -->------------------*---------+
+      //   |   iand                         |         |
+      //   |     iloadi <isClassFlags>      |         |
+      //   |       aloadi <componentClass>  |         |
+      //   |         aloadi <vft-symbol>    |         |
+      //   |           <array-reference>    |         |
+      //   |     iconst J9ClassIsValueType  |         |
+      //   |   iconst 0                     |         |
+      //   | BBEnd                          |         |
+      //   +--------------------------------+         |
+      //   | BBStart (Extension)            |         |
+      //   | NULLCHK                        |         |
+      //   |   Passthrough                  |         |
+      //   |     <value-reference>          |         |
+      //   | BBEnd                          |         |
+      //   +--------------------------------+         |
+      //                   |                          |
+      //                   +--------------------------+
+      //                   |
+      //                   v
+      //   +--------------------------------+
+      //   | BBStart                        |
+      //   | ArrayStoreCHK                  |
+      //   |   astorei/awrtbari             |
+      //   |     aladd                      |
+      //   |       aload <array>            |
+      //   |       index-offset-calculation |
+      //   |     aload <value>              |
+      //   +--------------------------------+
+      //
+      TR::SymbolReference *vftSymRef = comp()->getSymRefTab()->findOrCreateVftSymbolRef();
+      TR::SymbolReference *arrayCompSymRef = comp()->getSymRefTab()->findOrCreateArrayComponentTypeSymbolRef();
+      TR::SymbolReference *classFlagsSymRef = comp()->getSymRefTab()->findOrCreateClassFlagsSymbolRef();
+
+      TR::Node *vft = TR::Node::createWithSymRef(node, TR::aloadi, 1, anchoredArrayTT->getNode()->getFirstChild(), vftSymRef);
+      TR::Node *arrayCompClass = TR::Node::createWithSymRef(node, TR::aloadi, 1, vft, arrayCompSymRef);
+      TR::Node *loadClassFlags = TR::Node::createWithSymRef(node, TR::iloadi, 1, arrayCompClass, classFlagsSymRef);
+      TR::Node *isValueTypeNode = TR::Node::create(node, TR::iand, 2, loadClassFlags, TR::Node::iconst(node, J9ClassIsValueType));
+
+      TR::Node *ifNode = TR::Node::createif(TR::ificmpeq, isValueTypeNode, TR::Node::iconst(node, 0));
+      ifNode->copyByteCodeInfo(node);
```

Comment:
Is this needed ? I thought a node creation routine for a node with children would automatically copy in the first child's bytecode info.

Reply:
Thanks for catching that!  I hadn't realized that the first child would be used as the "originating bytecode node".  Fixed in pull request #12126